src/logic/calculate.js

```diff
@@ -0,0 +1,70 @@
+import operates from './operate';
+
+const calculate = (calcData, btnName) => {
+  let {
+    total,
+    next,
+    operate,
+  } = calcData;
+
+  const operators = ['+', '-', 'x', '÷'];
+  const numbers = ['0', '1', '3', '4', '5', '6', '7', '8', '9'];
+
+  switch (btnName) {
+    case 'AC':
+      total = null;
+      next = null;
+      operate = null;
+      break;
+
+    case '+/-':
+      if (total) {
+        total *= -1;
+      }
+      if (next) {
+        next *= -1;
+      }
+      break;
+
+    case '%':
+      if (total) {
+        next = (total * 0.01).toString;
+      }
+      break;
+
+    case '=':
+      if (total && next && operate) {
+        total = operates(total, next, operate);
+        next = null;
+        operate = null;
+      }
+      break;
+
+    case '.':
+      if (total) {
+        if (total.includes('.')) {
+          total = '0.';
+        }
+      }
```

Comment:
- [ ] This case wouldn't actually work well, because if `total` has an example value of `2.` then it would return `0.` which is wrong. Perhaps change the conditional to simply add a decimal to `total` if it doesn't already have a decimal and if it does it should be left as it is.